core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -67,6 +67,15 @@ class KafkaZkClient private (zooKeeperClient: ZooKeeperClient, isSecure: Boolean
   // Only for testing
   private[kafka] def currentZooKeeper: ZooKeeper = zooKeeperClient.currentZooKeeper
 
+  /*
+  * This variable holds the Zookeeper session id at the moment the ZooKeeperClient object
+  * is created and the subsequent updates of the session id, it is possible that the session id change
+  * over the time for 'Session expired', for instance.
+  * */
+  private var previousZooKeeperSessionId = zooKeeperClient.sessionId
+  // Only for testing, to simulate the ZooKeeper session id has changed
+  private var currentZooKeeperSessionId: Long = -1
```

Comment:
Do we really need this? Could we just create a subclass of KafkaZkClient and overwrite the return value of shouldReCreateEphemeralZNode() in the test?

Reply:
As you pointed out, this is a momentaneous workaround to tackle the issue. I have thought on a more elegant way to do it, as you propose. Probably am opting for the simples but not elegant solution for this. The rationale behind my proposal is:

- Keeping everything in the same place/file (now with the comments you suggested is better), in this way, the workaround is framed to a single place.
- Once the Zookeeper issue is addressed properly, we just have to come to this file/class and delete this code.

Having said that, am more than happy to change it if you think is better :)